src/test/ui/type-alias-impl-trait/defining-use-submodule.rs

```diff
@@ -0,0 +1,27 @@
+// run-pass
+
+#![feature(type_alias_impl_trait)]
+
+// test that the type alias impl trait defining use is in a submodule
+
+fn main() {
```

Comment:
I don't think this actually needed. Could just be a check-pass test?

Reply:
Yes, was just following the intention of the original test https://github.com/rust-lang/rust/pull/88172/files#diff-f0138a92e0c5bb0a679a72ff03ec1c30cf35ab5cc36f1143efdda3c98ad91bd3L14
But agreed with you that we could just make a check-pass test instead

Reply:
Done!